### PR TITLE
Added Encode/Decode traits (and new_element helper)

### PR DESCRIPTION
**Pull request by ryankurte**
Traits for encoding and decoding of SVD objects <-> XML Elements towards #46.
@Emilgardis I think this should be compatible with your approach to error handling?

### CODE REVIEW

**Comment by Emilgardis** [comment]
Yes it should work fine, however I don't see the point in this pr. I think adding everything together makes more sense.

I can redo my work on error-handling, that's fine.

File: src/types.rs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+use std::collections::HashMap;
+use xmltree::Element;
+
+/// Parse trait allows SVD objects to be parsed from XML/SVD elements.
+pub trait Parse {
+    /// Object returned by parse method
+    type Object;
+    /// Parsing error
+    type Error;
+    /// Parse an XML/SVD element into an SVD Object.
```
Comment (Emilgardis) [inline src/types.rs]: Sentence doesn't feel right to me. Maybe 

> [...] into it's corresponding `Object`

File: src/types.rs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    fn parse(&Element) -> Result<Self::Object, Self::Error>;
+}
+
+/// Encode trait allows SVD objects to be encoded into XML/SVD elements.
+pub trait Encode {
+    /// Encoding error
+    type Error;
+    /// Encode an SVD object into an XML element
```
Comment (Emilgardis) [inline src/types.rs]: Same here as above. Maybe just skip "an SVD object".

File: src/types.rs [hunk 3]
```diff
@@ -3,21 +3,21 @@
...
 
-/// Parse trait allows SVD objects to be parsed from XML/SVD elements.
+/// Parse trait allows SVD objects to be parsed from XML elements.
 pub trait Parse {
     /// Object returned by parse method
     type Object;
     /// Parsing error
     type Error;
-    /// Parse an XML/SVD element into an SVD Object.
+    /// Parse an XML/SVD element into it's corresponding Object.
```
Comment (Emilgardis) [inline src/types.rs]: Object refers to the trait type, put into code block. i.e `` `Object` ``

**Comment by Emilgardis** [comment]
I still feel uneasy about merging this without anything related to it. Maybe it would actually be easier to open a refactor branch and just directly push to it and then merge it, as you first suggested.

**Comment by ryankurte** [comment]
If you'd like I could update a couple of objects and open dependent PRs as
well so we can test / prove the interface? Then it's easy to merge em also.

**Comment by Emilgardis** [comment]
I say yes to implementing this on some objects, but no to opening dependent PRs. If we are going to make multiple small PRs depending on this I think skipping the PR system is the best choice.

**Comment by ryankurte** [comment]
Closed in favour of [upstream/refactor](https://github.com/japaric/svd/tree/refactor)

**Pull request closed without merging**